File: sdk-py/test/test_find_tools_remote.py
#!/usr/bin/env python3
"""
test_find_tools_remote.py

CLI test for find_tools_remote.
Usage:
  python test_find_tools_remote.py [--http|--mcp] [--repo=<id>] "<prompt>"

Exit codes: 0 = tools found, 1 = error, 2 = no tools found
"""

from __future__ import annotations

import json
import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).parent.parent / "src"))
from find_tools_remote import RemoteMethod, find_tools_remote, _parse_args

# ─── CLI ──────────────────────────────────────────────────────────────────────

def _main(argv: list[str]) -> int:
    args = _parse_args(argv)

    if args is None:
        print(
            "Usage: python test_find_tools_remote.py [--http|--mcp] [--repo=<id>] <prompt>",
            file=sys.stderr,
        )
        return 1

    match args.method:
        case RemoteMethod.MCP:
            print("[method: MCP]", file=sys.stderr)
        case RemoteMethod.HTTP:
            print(f"[method: HTTP API, repo: {args.repo_id}]", file=sys.stderr)

    print(f'\nPrompt : "{args.prompt}"')
    print("─" * 60)

    try:
        results = find_tools_remote(args.prompt, args.method, args.repo_id)
    except Exception as exc:
        print(f"Error: {exc}", file=sys.stderr)
        return 1

    match results:
        case []:
            print("No matching tools found.")
            return 2
        case tools:
            print(f"Found {len(tools)} tool(s):\n")
            print(json.dumps([t.to_dict() for t in tools], indent=2))
            return 0


if __name__ == "__main__":
    sys.exit(_main(sys.argv[1:]))

File: sdk-ts/test/test_find_tools_remote.ts
/**
 * test_find_tools_remote.ts
 *
 * CLI test for find_tools_remote.
 * Usage:
 *   ts-node test_find_tools_remote.ts --http [--repo=<id>] "<prompt>"
 *   ts-node test_find_tools_remote.ts --mcp "<prompt>"
 *
 * Exit code 0 = tools found, exit code 2 = no tools found, exit code 1 = error.
 */

import {
  findToolsRemoteHttp,
  findToolsRemoteMcp,
} from "../src/find_tools_remote";

interface MatchedTool {
  title: string;
  description: string;
  url: string;
  image: { small: string; large: string };
}

async function runRemoteTest(
  prompt: string,
  method: "http" | "mcp",
  repoId: string
): Promise<void> {
  console.log(`\nPrompt : "${prompt}"`);
  console.log(`Method : ${method}${method === "http" ? ` (repo: ${repoId})` : ""}`);
  console.log("─".repeat(60));

  let results: MatchedTool[];
  try {
    if (method === "mcp") {
      results = await findToolsRemoteMcp(prompt);
    } else {
      results = await findToolsRemoteHttp(prompt, repoId);
    }
  } catch (err: unknown) {
    const message = err instanceof Error ? err.message : String(err);
    console.error(`Request failed: ${message}`);
    process.exit(1);
  }

  if (results.length === 0) {
    console.log("No matching tools found.");
    process.exit(2);
  }

  console.log(`Found ${results.length} tool(s):\n`);
  console.log(JSON.stringify(results, null, 2));
  process.exit(0);
}

// ─── Parse CLI arguments ──────────────────────────────────────────────────────

const args = process.argv.slice(2);

let method: "http" | "mcp" = "http";
let repoId = "default";
const promptParts: string[] = [];

for (const arg of args) {
  if (arg === "--http") {
    method = "http";
  } else if (arg === "--mcp") {
    method = "mcp";
  } else if (arg.startsWith("--repo=")) {
    repoId = arg.split("=")[1];
  } else {
    promptParts.push(arg);
  }
}

const prompt = promptParts.join(" ");

if (!prompt) {
  console.error(
    "Usage: ts-node test_find_tools_remote.ts [--http|--mcp] [--repo=<id>] <prompt>"
  );
  process.exit(1);
}

runRemoteTest(prompt, method, repoId);
